sdk/python/tests/compiler/testdata/recur_nested_pipelineloop_cr2.yaml

```diff
@@ -0,0 +1,142 @@
+metadata:
+  name: nested-recursion-pipeline-graph-flip-component-b-3
+spec:
+  pipelineSpec:
+    params:
+    - name: just_one_iteration
+      type: string
+    - name: flip-coin-output
+      type: string
+    - name: maxVal
+      type: string
+    tasks:
+    - name: print
+      params:
+      - name: flip-coin-output
+        value: $(params.flip-coin-output)
+      taskSpec:
+        steps:
+        - name: main
+          command:
+          - echo
+          - $(inputs.params.flip-coin-output)
+          image: alpine:3.6
+        params:
+        - name: flip-coin-output
+          type: string
+        metadata:
+          labels:
+            pipelines.kubeflow.org/pipelinename: ''
+            pipelines.kubeflow.org/generation: ''
+            pipelines.kubeflow.org/cache_enabled: "true"
+          annotations:
+            tekton.dev/template: ''
+      when:
+      - input: $(tasks.condition-4.results.outcome)
+        operator: in
+        values:
+        - "true"
+      timeout: 0s
+    - name: flip-coin-2
+      taskSpec:
+        steps:
+        - name: main
+          args:
+          - python -c "import random; result = 'heads' if random.randint(0,1) == 0
+            else 'tails'; print(result)" | tee $(results.output.path)
+          command:
+          - sh
+          - -c
+          image: python:alpine3.6
+        results:
+        - name: output
+          description: /tmp/output
+        metadata:
+          labels:
+            pipelines.kubeflow.org/pipelinename: ''
+            pipelines.kubeflow.org/generation: ''
+            pipelines.kubeflow.org/cache_enabled: "true"
+          annotations:
+            tekton.dev/template: ''
+      when:
+      - input: $(tasks.condition-4.results.outcome)
+        operator: in
+        values:
+        - "true"
+      runAfter:
+      - print
+      timeout: 0s
+    - name: condition-4
+      params:
+      - name: operand1
+        value: $(params.flip-coin-output)
+      - name: operand2
+        value: heads
+      - name: operator
+        value: ==
+      taskSpec:
+        results:
+        - name: outcome
+          description: Conditional task outcome
+        params:
+        - name: operand1
+          type: string
+        - name: operand2
+          type: string
+        - name: operator
+          type: string
+        steps:
+        - script: |-
+            python -c 'import sys
+            input1=str.rstrip(sys.argv[1])
+            input2=str.rstrip(sys.argv[2])
+            try:
+              input1=int(input1)
+              input2=int(input2)
+            except:
+              input1=str(input1)
+            outcome="true" if (input1 $(inputs.params.operator) input2) else "false"
+            f = open("/tekton/results/outcome", "w")
+            f.write(outcome)
+            f.close()' '$(inputs.params.operand1)' '$(inputs.params.operand2)'
+          image: python:alpine3.6
+      when:
+      - input: $(tasks.condition-2.results.outcome)
```

Comment:
This when expression is not necessary because condition-2 is not in this pipeline loop. If the parent condition is not inside this graph, then we don't need to add it here.

Reply:
I added a quick fix here that should solve this problem
https://github.com/pugangxa/kfp-tekton/pull/1

Reply:
Thanks Tommy, it's fine.

Reply:
thanks, you will need to recompile this test in order to remove this when expression